products/views.py

```diff
@@ -1,9 +1,82 @@
-from django.shortcuts import render
-from .models import Brand, Category, Product
-from django.contrib import messages
 from django.db.models import Count
+from django.shortcuts import render, redirect, get_object_or_404
+from .forms import AddCategoryForm
+from .models import Category, Brand, Product
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.contrib import messages
+from products.headings import AdminPortalHeadings
 from django.contrib.auth.decorators import login_required
-from .headings import AdminPortalHeadings
+
+
+# Create your views here.
+@login_required
+def add_category(request):
+    """ this view is useful to add category """
+
+    context = {}
+    if request.method == "POST":
+        category = AddCategoryForm(request.POST, request.FILES)
+        if category.is_valid():
+            category.save()
+            messages.success(request, AdminPortalHeadings.PRODUCT_ADDED)
+        return redirect('view-category')
+
+    category = AddCategoryForm()
+    context['form'] = category
+    context['heading'] = '  Add Categories'
+    return render(request, "product/category/add_category.html", context)
+
+
+@login_required
+def update_category(request, pk):
+    """ this view is useful for update product category """
+
+    context = {}
+    if request.method == "POST":
```

Comment:
@Yshah112 instead of writing get_object_or_404(Category, pk=pk) multiple times, just assign it to a variable and use it wherever it is needed.

cc: @lalitvasoya

Reply:
done